src/ImageSharp/Common/Helpers/Vector4Utilities.cs

```diff
@@ -11,8 +11,20 @@ namespace SixLabors.ImageSharp
     /// <summary>
     /// Utility methods for the <see cref="Vector4"/> struct.
     /// </summary>
-    internal static class Vector4Utils
+    internal static class Vector4Utilities
     {
+        /// <summary>
+        /// Restricts a vector between a minimum and a maximum value.
+        /// 5x Faster then <see cref="Vector4.Clamp(Vector4, Vector4, Vector4)"/>.
+        /// </summary>
+        /// <param name="x">The vector to restrict.</param>
+        /// <param name="min">The minimum value.</param>
+        /// <param name="max">The maximum value.</param>
+        /// <returns>The <see cref="Vector4"/>.</returns>
+        [MethodImpl(InliningOptions.ShortMethod)]
+        public static Vector4 FastClamp(Vector4 x, Vector4 min, Vector4 max)
```

Comment:
Just a note that I've also called out on https://github.com/dotnet/runtime/issues/34017.

Due to how HLSL/GLSL behave and how the underlying `minps`/`maxps` instructions behave, you may need to be careful on the ordering of `x, y` here.

For regular inputs, the order won't matter and you'll get the correct/expected result. However, if either input is `NaN` or both inputs are `Zero`, the result returned might differ based on how you order `x, y`.

Reply:
I finished the analysis and it looks like you want `Vector4.Min(Vector4.Max(x, min), max)` (so just swap `min` and `x`): https://github.com/dotnet/runtime/issues/34017#issuecomment-603524623

Reply:
@tannergooding thanks a lot for the engagement!

Reply:
Ah @tannergooding this is fantastic! It explains the issue I was seeing when attempting to replace a part within the fallback intrinsics. The change fixes that.